### PR TITLE
feat(EMS-2433): no PDF - Eligibility - No short term cover redirect

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction :pencil2:

This PR adds a redirect for when `noShortTermCover` is available for the buyer country but `insuranceSupport` is available

## Resolution :heavy_check_mark:

* Added logic to check if `noShortTermCover` and if `noinsuranceSupport`, then to redirect to `CONTRACT_TOO_SHORT` page
* Updated `canApplyForInsuranceOnline` to take boolean with shortTermCover
* Removed mapping for applying offline
* Removed all instances of `canApplyForInsuranceOffline`
* Removed redirect to APPLY_OFFLINE from buyer country controller
* Updated tests
* Added e2e tests for unsupported and contract too short countries
* Remove now irrelevant "cannot apply offline" exit page

### CODE REVIEW

File: e2e-tests/insurance/cypress/e2e/journeys/eligibility/buyer-country/buyer-country-no-short-term-cover.spec.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      cy.keyboardInput(autoCompleteField(FIELD_ID).input(), COUNTRY_NAME);
+
+      const results = autoCompleteField(FIELD_ID).results();
+      results.first().click();
+
+      cy.clickSubmitButton();
+    });
+
+    it('redirects to `cannot apply` exit page', () => {
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/eligibility/buyer-country/buyer-country-no-short-term-cover.spec.js:40]:
```js
it(`should redirect to ${CONTRACT_TOO_SHORT}`, () => ...
```
?

File: src/api/helpers/map-CIS-countries/map-CIS-country/index.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export const mapCisCountry = (country: CisCountry): MappedCisCountry => {
   mapped.canGetAQuoteByEmail = canGetAQuoteByEmail(mapped);
   mapped.cannotGetAQuote = cannotGetAQuote(mapped);
 
-  mapped.canApplyForInsuranceOnline = applyForInsuranceOnline(country.shortTermCoverAvailabilityDesc, mapped.riskCategory);
+  mapped.canApplyForInsuranceOnline = applyForInsuranceOnline(mapped.shortTermCover, mapped.riskCategory);
```
Comment (ttbarnes) [inline src/api/helpers/map-CIS-countries/map-CIS-country/index.ts:32]: Is `shortTermCoverAvailabilityDesc` no longer relevant here - is `shortTermCover` the mapping result of `shortTermCoverAvailabilityDesc`?

File: src/ui/server/controllers/insurance/eligibility/buyer-country/index.test.ts [hunk 3]
```diff
@@ -202,6 +202,41 @@ describe('controllers/insurance/eligibility/buyer-country', () => {
...
+        expect(req.session.submittedData).toEqual(expected);
+      });
+
+      it(`should redirect to ${CONTRACT_TOO_SHORT}`, async () => {
+        await post(req, res);
+
+        expect(res.redirect).toHaveBeenCalledWith(CONTRACT_TOO_SHORT);
+      });
+    });
+
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/eligibility/buyer-country/index.test.ts:239]: Not a blocker, could do with splitting up this unit test into different files for different scenarios

File: src/ui/server/controllers/insurance/eligibility/buyer-country/index.ts [hunk 4]
```diff
@@ -91,6 +91,23 @@ export const post = async (req: Request, res: Response) => {
       return res.redirect(CANNOT_APPLY_ROUTE);
     }
 
+    /**
+     * If a country has no insurance support and no short term cover,
+     * redirect to contract too short page.
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/eligibility/buyer-country/index.ts:96]: redirect to `CONTRACT_TOO_SHORT`?

File: src/api/helpers/map-CIS-countries/map-CIS-country/index.ts (hunk 2 above)
Comment (Zainzzkk) [reply src/api/helpers/map-CIS-countries/map-CIS-country/index.ts:32]: yes it is mapped on line 23

File: e2e-tests/commands/insurance/eligibility/assert-exit-page-url-buyer-country.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * assertExitPageUrlBuyerCountry
+ * can assert if url is correct after entering country that leads to exit page
+ * @param {String} country to input
+ * @param {String} url to check for exit page
+ */
+const assertExitPageUrlBuyerCountry = (country, url) => {
+  cy.keyboardInput(autoCompleteField(FIELD_ID).input(), country);
+
+  const results = autoCompleteField(FIELD_ID).results();
+  results.first().click();
```
Comment (ttbarnes) [inline e2e-tests/commands/insurance/eligibility/assert-exit-page-url-buyer-country.js:18]: Could use `cy.autocompleteKeyboardInput` here

Comment (ttbarnes) [reply e2e-tests/commands/insurance/eligibility/assert-exit-page-url-buyer-country.js:18]: Can we rename this command, as it's not just asserting the URL?

File: e2e-tests/fixtures/countries.js [hunk 6]
```diff
@@ -53,6 +53,26 @@ const mockCountries = [
...
+    ISO_CODE: 'BLR',
+  },
+  {
+    NAME: 'Russia',
+    ISO_CODE: 'RUS',
+  },
+  {
+    NAME: 'Burma',
+    ISO_CODE: 'MMR',
+  },
```
Comment (ttbarnes) [inline e2e-tests/fixtures/countries.js:75]: Can we order these alphabetically by `NAME`?

File: e2e-tests/insurance/cypress/e2e/journeys/eligibility/buyer-country/buyer-country-unsupported-countries.spec.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  START,
+  ELIGIBILITY: { CANNOT_APPLY },
+} = INSURANCE_ROUTES;
+
+const COUNTRY_NAME_1 = COUNTRY_APPLICATION_SUPPORT.UNSUPPORTED_1.NAME;
+const COUNTRY_NAME_2 = COUNTRY_APPLICATION_SUPPORT.UNSUPPORTED_2.NAME;
+const COUNTRY_NAME_3 = COUNTRY_APPLICATION_SUPPORT.UNSUPPORTED_3.NAME;
+const COUNTRY_NAME_4 = COUNTRY_APPLICATION_SUPPORT.UNSUPPORTED_4.NAME;
+
+context('Insurance - Buyer country page - Unsupported countries', () => {
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/eligibility/buyer-country/buyer-country-unsupported-countries.spec.js:14]: User story?

File: e2e-tests/insurance/cypress/e2e/journeys/eligibility/buyer-country/buyer-country-unsupported-country-page.spec.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+} = INSURANCE_ROUTES;
+
+const FIELD_ID = FIELD_IDS.ELIGIBILITY.BUYER_COUNTRY;
+
+const COUNTRY_NAME = COUNTRY_APPLICATION_SUPPORT.UNSUPPORTED_1.NAME;
+
+const baseUrl = Cypress.config('baseUrl');
+
+context(
+  'Insurance - Buyer country page - as an exporter, I want to check if UKEF offer credit insurance policy for where my buyer is based - submit unsupported country',
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/eligibility/buyer-country/buyer-country-unsupported-country-page.spec.js:22]: typo RE page name

File: e2e-tests/insurance/cypress/e2e/journeys/apply-offline-page.spec.js [hunk 9]
```diff
@@ -1,51 +1,26 @@
...
-    const results = autoCompleteField(FIELD_ID).results();
-    results.first().click();
-
-    cy.clickSubmitButton();
-
-    const expectedUrl = `${baseUrl}${APPLY_OFFLINE}`;
-
-    cy.assertUrl(expectedUrl);
+    cy.navigateToUrl(applyOfflineUrl);
+    cy.assertUrl(applyOfflineUrl);
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/e2e/journeys/apply-offline-page.spec.js:16]: adam said there is one part of the eligibility for now where we need it, so have kept it in

File: src/api/helpers/map-CIS-countries/map-CIS-country/no-insurance-support/index.test.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+describe('helpers/map-CIS-countries/map-CIS-country/no-insurance-support', () => {
+  describe(`when marketRiskAppetitePublicDesc is ${NO_COVER}`, () => {
+    it('should return true', () => {
+      const result = noInsuranceSupportAvailable(NO_COVER);
+
+      expect(result).toEqual(true);
+    });
+  });
+
+  describe(`when originalShortTermCover is NOT '${noInsuranceSupportAvailable}'`, () => {
```
Comment (ttbarnes) [inline src/api/helpers/map-CIS-countries/map-CIS-country/no-insurance-support/index.test.ts:17]: Looks like `noInsuranceSupportAvailable` should be `NO_COVER`

File: src/ui/server/controllers/insurance/eligibility/buyer-country/countries.test.ts [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+import mapSubmittedEligibilityCountry from '../../../../helpers/mappings/map-submitted-eligibility-country';
+import { updateSubmittedData } from '../../../../helpers/update-submitted-data/insurance';
+import { Country, Request, Response } from '../../../../../types';
+import { mockReq, mockRes, mockCountries } from '../../../../test-mocks';
+
+const {
+  ELIGIBILITY: { CANNOT_APPLY: CANNOT_APPLY_ROUTE, TOTAL_VALUE_INSURED, BUYER_COUNTRY_CHANGE, CHECK_YOUR_ANSWERS, CONTRACT_TOO_SHORT },
+} = INSURANCE_ROUTES;
+
+describe('controllers/insurance/eligibility/buyer-country', () => {
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/eligibility/buyer-country/countries.test.ts:16]: Should we rename this file`country-redirects`?

And add ` - redirects` to the `describe` here?

File: src/api/helpers/map-CIS-countries/map-CIS-country/no-insurance-support/index.test.ts (hunk 10 above)
Comment (Zainzzkk) [reply src/api/helpers/map-CIS-countries/map-CIS-country/no-insurance-support/index.test.ts:17]: whoops

File: e2e-tests/insurance/cypress/e2e/journeys/apply-offline-page.spec.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (ttbarnes) [reply e2e-tests/insurance/cypress/e2e/journeys/apply-offline-page.spec.js:16]: What journey is that?

File: src/ui/server/constants/routes/insurance/index.ts [hunk 12]
```diff
@@ -45,7 +45,6 @@ export const INSURANCE_ROUTES = {
     NEED_TO_START_AGAIN: `${INSURANCE_ROOT}${ELIGIBILITY_ROOT}/need-to-start-again`,
     CONTRACT_TOO_SHORT: `${INSURANCE_ROOT}${ELIGIBILITY_ROOT}/contract-too-short`,
   },
-  APPLY_OFFLINE: `${INSURANCE_ROOT}/apply-using-our-form`,
```
Comment (ttbarnes) [inline src/ui/server/constants/routes/insurance/index.ts:48]: Still some instances of `APPLY_OFFLINE` in template constants and content strings, can these be removed now?

Comment (ttbarnes) [reply src/ui/server/constants/routes/insurance/index.ts:48]: I'll address this after merge

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**